packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -225,6 +226,45 @@ export function useInputHandler<T extends ValidatableFormElement>(
     }
 
     if (!isSSR) {
+        /// Propagates any custom constraint validation message to the underlying DOM element
+        const updateCustomValidationMessage = (): void => {
+            if (!(props.useHtml5Validation ?? true)) {
+                return;
+            }
+            const element = maybeElement.value;
+            if (!isDefined(element)) {
+                return;
+            }
+            const validity = props.customValidity ?? "";
+            if (typeof validity === "string") {
+                element.setCustomValidity(validity);
+            } else {
+                element.setCustomValidity(validity(element.validity));
+            }
+        };
+        // Note that using watchEffect will implicitly pick up any reactive dependencies used
+        // inside props.customValidity, which should help the computed message stay up to date.
+        watchEffect(updateCustomValidationMessage);
```

Comment:
Have you tested if the props are really a reactive object with can watchEffect look at? I'm not sure about that, but i could be.

Reply:
Based on the call sites I've looked at, components that use the composable seem to pass their own `props` objects directly through, so the object itself will be reactive. More importantly, any reactive dependencies inside a `watchEffect` function [are automatically registered](https://vuejs.org/api/reactivity-core.html#watcheffect), so if `props.customValidity` has an internal dependency on a reactive value (e.g. it reads from a `ref`), that will get registered against the watcher as well. This seems to work in practice — with the caveat that if `props.customValidity` has any conditional dependencies (e.g. it only reads a `ref` if another condition is true), those won't get registered unless they're observed _within_ the `watchEffect` context.